vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/GridMultiSelectionModel.java

```diff
@@ -41,13 +41,17 @@
      * only visible if an in-memory data provider is used
      * {@link DataProvider#isInMemory()}.
      */
-    public enum SelectAllCheckboxVisibility {
+    enum SelectAllCheckboxVisibility {
 
         /**
-         * Shows the select all checkbox, regardless of data provider used.
+         * Shows the select all checkbox, if the in-memory or defined size lazy
+         * data provider is used.
```

Comment:
Maybe instead simplify (no need to talk about data provider anymore) :
"
Shows the select all checkbox, if in-memory data is used. For lazy data, the checkbox is only shown when a count callback has been provided. For lazy data with unknown count, the checkbox will never be shown.

For lazy data, allowing the user to select all will result ...
"

Reply:
Done